src/Fmt_ast.ml

```diff
@@ -1603,24 +1603,38 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
         $ fmt_atrs )
   | Pexp_open (flag, {txt; loc}, e0) ->
       let override = Poly.(flag = Override) in
+      (* open_style = `Long means we use let open Module in (...) style;
+         open_style = `Short means we use Module.(...) style; open_style =
+         `Auto means we use the one fitting best. *)
+      let open_style =
+        match c.conf.open_style with
+        | `Auto -> `Auto
+        | `Short -> `Short
+        | `Preserve ->
+          match Source.string_between c.source loc e0.pexp_loc with
```

Comment:
It doesn't work with the following example:
```
let _ =
  A.(* in *)((* in *)x y)
```
String.between returns `.(* in *)((* in *)`.

Have a look at what's done for Ppat_interval using `Source.tokens_at`